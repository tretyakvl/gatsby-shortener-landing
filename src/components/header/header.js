import React from 'react'
import PropTypes from 'prop-types'
import { StaticQuery, graphql, Link } from 'gatsby'

import Navigation from './navigation/navigation'

import style from './header.module.css'
import Logo from '../logo/logo'

export const PureHeader = ({ navigation }) => {
  return (
    <header className={style.header}>
      <Link to='/'>
        <Logo />
      </Link>
      <Navigation navigation={navigation} />
    </header>
  )
}

export default props => (
  <StaticQuery
    query={graphql`
      query {
        headerYaml {
          navigation {
            primary
            login
          }
        }
      }
    `}
    render={({ headerYaml }) => <PureHeader {...headerYaml} {...props} />}
  />
)

PureHeader.propTypes = {
  navigation: PropTypes.shape({
    login: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)),
    primary: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string))
  }).isRequired
}
