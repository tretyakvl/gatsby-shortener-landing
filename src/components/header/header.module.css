.header {
  position: relative;

  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  max-width: var(--max-width);
  height: 6.875rem;
  margin: var(--margin);
  padding: var(--padding);
  padding-top: 0;
  padding-bottom: 0;
}

@media (min-width: 40rem) {
  .header {
    height: 8.5rem;
  }
}
