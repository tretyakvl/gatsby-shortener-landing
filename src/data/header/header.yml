navigation: 
  primary:
    - ['Features', '/features']
    - ['Pricing', '/pricing']
    - ['Resources', '/resources']
  login:
    - ['Login', '/login']
    - ['Sing Up', '/sing-up']