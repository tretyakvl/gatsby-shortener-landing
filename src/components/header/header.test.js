import React from 'react'
import renderer from 'react-test-renderer'
import { PureHeader as Header } from './header'

describe('Header', () => {
  it('renders correctly', () => {
    const data = {
      primary: [
        ['About', '/about'],
        ['Team', '/team'],
        ['Price', '/price']
      ],
      login: [
        ['Login', '/login'],
        ['Register', '/register']
      ]
    }

    const tree = renderer
      .create(<Header navigation={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
