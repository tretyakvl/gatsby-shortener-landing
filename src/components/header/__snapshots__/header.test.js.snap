// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header"
>
  <a
    href="/"
  >
    <test-file-stub
      className="undefined"
    />
  </a>
  <nav
    className="navMobile"
  >
    <button
      className="hamburger"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        height="25"
        viewBox="0 0 25 25"
        width="25"
      >
        <path
          d="M 2 4 L 23 4"
          fill="transparent"
          stroke="var(--gray-500)"
          strokeWidth="3"
          style={Object {}}
        />
        <path
          d="M 2 12.5 L 23 12.5"
          fill="transparent"
          opacity={1}
          stroke="var(--gray-500)"
          strokeWidth="3"
          style={Object {}}
        />
        <path
          d="M 2 21 L 23 21"
          fill="transparent"
          stroke="var(--gray-500)"
          strokeWidth="3"
          style={Object {}}
        />
      </svg>
      <span
        className="visually-hidden"
      >
        Open/close menu
      </span>
    </button>
    <div
      className="navMobile__dropdown"
      style={
        Object {
          "clipPath": "circle(0% at 110% -10%)",
        }
      }
    >
      <ul
        className=""
      >
        <li>
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="/team"
          >
            Team
          </a>
        </li>
        <li>
          <a
            href="/price"
          >
            Price
          </a>
        </li>
      </ul>
      <ul
        className=""
      >
        <li>
          <a
            href="/login"
          >
            Login
          </a>
        </li>
        <li>
          <a
            className="button-module button"
            href="/sign-up"
          >
            Sign Up
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
